rule.php

```diff
@@ -150,6 +150,18 @@ public static function validate_settings_form_fields(array $errors,
             }
         }
 
+        // Edge case for Use Browser Exam Keys and empty Allowed Browser Exam Keys.
+        $cases = [
+            settings_provider::USE_SEB_UPLOAD_CONFIG,
+            settings_provider::USE_SEB_CLIENT_CONFIG
+        ];
+        $requiresetting = $quizsettings->get('requiresafeexambrowser');
+        if (in_array($requiresetting, $cases)) {
```

Comment:
how about to abstract it to a method + cover with tests?

Reply:
Not sure about the best place to put it. Maybe to settings_provider?

Reply:
Also I think we need to empty allowedbrowserexamkeys in quiz_settings as we do with templateid. just to make sure we don't save hidden value from the form